PowerEditor/src/ScitillaComponent/ScintillaEditView.cpp

```diff
@@ -2414,9 +2414,9 @@ void ScintillaEditView::convertSelectedTextTo(bool Case)
 		for (int i = 0 ; i < nbChar ; ++i)
 		{
 			if (Case == UPPERCASE)
-				selectedStrW[i] = (WCHAR)::CharUpperW((LPWSTR)selectedStrW[i]);
+				selectedStrW[i] = (WCHAR)(UINT_PTR)::CharUpperW((LPWSTR)selectedStrW[i]);
 			else
-				selectedStrW[i] = (WCHAR)::CharLowerW((LPWSTR)selectedStrW[i]);
+				selectedStrW[i] = (WCHAR)(UINT_PTR)::CharLowerW((LPWSTR)selectedStrW[i]);

```

Comment:
You can do static_cast to wchar_t.
But the the whole code uses C-stlyle casts, I didn't want to change it drastically.

```cpp
return static_cast<wchar_t>(reinterpret_cast<UINT_PTR>(CharUpper(reinterpret_cast<LPWSTR>(c))));
```